### PR TITLE
Merge 0.7.x into 0.8.x

### PR DESCRIPTION
**Pull request by wrijeff**
*Description of changes:*
* Merging 0.7.x changes to ensure nothing gets missed in the 0.8.0 release. 
  * Mostly documentation and build changes.
-----

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by kowshikn** [comment]
/s/processor/prepper in the readme?

**Comment by wrijeff** [comment]
> /s/processor/prepper in the readme?

Ah, good catch. This file wasn't in the original processor/prepper swap, will fix

File: release/docker/opendistroforelasticsearch-data-prepper-0.7.1-alpha.Dockerfile [hunk 1]
```diff
@@ -1,6 +1,4 @@
-# building on CentOS 7 to be in line with
-# https://github.com/opendistro-for-elasticsearch/opendistro-build/blob/master/elasticsearch
-
+# For the v0.7.0-alpha release we will use the amazoncorretto amazon linux.
 FROM amazoncorretto:15-al2-full
 ARG CONFIG_FILEPATH
 ARG JAR_FILE
```
Comment (yadavcbala) [inline release/docker/opendistroforelasticsearch-data-prepper-0.7.1-alpha.Dockerfile:4]: should we update this `CMD java $JAVA_OPTS -jar data-prepper.jar ${ENV_CONFIG_FILEPATH}` to also include the configuration file for DataPrepperServer ?

File: release/docker/opendistroforelasticsearch-data-prepper-0.7.1-alpha.Dockerfile [hunk 2]
```diff
@@ -1,6 +1,4 @@
-# building on CentOS 7 to be in line with
-# https://github.com/opendistro-for-elasticsearch/opendistro-build/blob/master/elasticsearch
-
+# For the v0.7.0-alpha release we will use the amazoncorretto amazon linux.
```
Comment (yadavcbala) [inline release/docker/opendistroforelasticsearch-data-prepper-0.7.1-alpha.Dockerfile:1]: update this comment to make it release neutral

**Review by yadavcbala — APPROVED** [review]

File: release/docker/opendistroforelasticsearch-data-prepper-0.7.1-alpha.Dockerfile (hunk 2 above)
Comment (wrijeff) [reply release/docker/opendistroforelasticsearch-data-prepper-0.7.1-alpha.Dockerfile:1]: Will remove it in a follow-up PR (the `FROM` line is self-explanatory)

File: release/docker/opendistroforelasticsearch-data-prepper-0.7.1-alpha.Dockerfile (hunk 1 above)
Comment (wrijeff) [reply release/docker/opendistroforelasticsearch-data-prepper-0.7.1-alpha.Dockerfile:4]: Probably not, 0.7.x didn't have server config files at the time

**Pull request merged**